File: PublishR/Models/Input.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PublishR.Models
{
    public class Input
    {
        [JsonProperty("type")]
        public string Type { get; set; }

        [JsonProperty("name")]
        public string Name { get; set; }

        [JsonProperty("label")]
        public string Label { get; set; }

        [JsonProperty("description")]
        public string Description { get; set; }

        [JsonProperty("hint")]
        public string Hint { get; set; }

        [JsonProperty("pattern")]
        public string Pattern { get; set; }

        [JsonProperty("required")]
        public bool Required { get; set; }

        [JsonProperty("range")]
        public Range Range { get; set; }

        [JsonProperty("length")]
        public Length Length { get; set; }

        [JsonProperty("value")]
        public object Value { get; set; }

        [JsonProperty("options")]
        public IList<Option> Options { get; set; }

        [JsonProperty("properties")]
        public IDictionary<string, object> Properties { get; set; }
    }
}

File: PublishR.Client/Models/Input.ts
module publishr.client {
    "use strict";

    export class Input {
        type: string;
        name: string;
        label: string;
        description: string;
        hint: string;
        pattern: string;
        required: boolean;
        range: Range;
        length: Length;
        value: any;
        options: Option[];
        properties: { [name: string]: any };
    }
} 